nndownload/nndownload.py

```diff
@@ -54,7 +54,11 @@
 TIMESHIFT_USE_URL = "https://live.nicovideo.jp/api/timeshift.ticket.use"
 TIMESHIFT_RESERVE_URL = "https://live.nicovideo.jp/api/timeshift.reservations"
 
-VALID_URL_RE = re.compile(r"(?:https?://(?:(?:(?:(ch|sp|www|seiga)\.)|(?:(live[0-9]?|cas)\.))?(?:(?:nicovideo\.jp\/(watch|mylist|user\/illust|user\/manga|user|comic|seiga|gate|article|channel|manga|illust)?)(?(3)\/|))|(nico\.ms)\/))(?:((?:(?:[a-z]{2})?[0-9]+)|[a-zA-z-0-9]+?)\/?)(?:\/(video|mylist|live|blomaga|list))?(?(6)\/((?:[a-z]{2})?[0-9]+))?(?:\?(?:user_id=(.*)|.*)?)?$")
```

Comment:
Before there was a weird bit in this regex like `[a-zA-z-0-9]+`, which doesn't really make sense to me. There's a range between uppercase A to lowercase z, then a dash (which I'm not really even sure what that will do) followed by another range.

I just changed it to `\w`, which should cover that. If you want to be super pedantic about replicating the regex, all it takes is flipping the `ascii` arg in the match

Reply:
It actually matches `-` literally which makes it look funny, some blogs have slugs using that.

Reply:
Ah I see. I'm still not sure if the `A-z` range is intended, that will match all the chars between capital A and lowercase z, as shown below in the ascii table.

```
...   40 @    50 P    60 `    70 p
...   41 A    51 Q    61 a    71 q
...   42 B    52 R    62 b    72 r
...   43 C    53 S    63 c    73 s
...   44 D    54 T    64 d    74 t
...   45 E    55 U    65 e    75 u
...   46 F    56 V    66 f    76 v
...   47 G    57 W    67 g    77 w
...   48 H    58 X    68 h    78 x
...   49 I    59 Y    69 i    79 y
...   4A J    5A Z    6A j    7A z
...   4B K    5B [    6B k    7B {
...   4C L    5C \    6C l    7C |
...   4D M    5D ]    6D m    7D }
...   4E N    5E ^    6E n    7E ~
...   4F O    5F _    6F o    7F DEL
```

I'll change it back (since \w doesn't seem to match hyphens):

![image](https://user-images.githubusercontent.com/28817893/123505651-a3bcba00-d6a3-11eb-8d80-3d92f70b5234.png)